Firestore/src/Query.php

```diff
@@ -766,6 +712,27 @@ private function buildPosition($key, $fieldValues, $before)
         ], true);
     }
 
+    /**
+     * Find fieldFilters in a composite filter.
+     *
+     * @param array $filter The composite filter.
+     * @return array array of field filters.
+     */
+    private function flattenFilter($filter)
```

Comment:
Add test for `flattenFilter`

Reply:
The `flattenFilter` is being used by `startAt` function in conjunction with the inequality operator in the `where` clause. Since it is a private function, we have covered the UTs for this `flattenFilter` by `testPositionInequalityWithFieldFilter()` and `testPositionInequalityWithCompositeFilter()` tests.